### DISCUSSION TITLE
Unexpected 'class does not exist' error when compiling in Tailwind 3.0

### DISCUSSION THREAD

**Discussion by danielbachhuber** [tailwindlabs/tailwindcss#6902] [Help] [answered]
I'm getting an unexpected 'class does not exist' error when compiling with Tailwind 3.0.10:

```
(93:3) /Users/danielbachhuber/projects/foodbloggerpro/wp-content/assets/scss/style.scss The `top-0` class does not exist. If `top-0` is a custom class, make sure it is defined within a `@layer` directive.

  91 | }
  92 | ol.square-list li:before {
> 93 |   @apply absolute top-0 text-white bg-grey text-center rounded;
     |   ^
  94 |   --size: 22px;
  95 |   left: calc(-1 * var(--size) - 10px);
```

I don't get the error with Tailwind 2.2.19.

Here's my `tailwind.config.js` for 2.2.9: https://gist.github.com/danielbachhuber/833c60d632a8410fec166a677c833a8d

Here's my `tailwind.config.js` for 3.0.10: https://gist.github.com/danielbachhuber/04d7d1ca1eefffbd27f75f88e369597a

Here's a repo that reproduces the issue: https://github.com/danielbachhuber/tailwind-class-does-not-exist-issue

To help me diagnose further...
1. What Tailwind code is responsible for generating the `top-0` class? I wasn't able to track it down in a 5 minute review of the source code.
2. Was there any related code that changed in Tailwind 3.0?

Thanks in advance!

**Comment by adamwathan** [comment, marked as answer]
Hey! It looks like it's because your `0` value in the `inset` key doesn't have quotes but I'm not sure why this matters or doesn't error or something.

https://github.com/danielbachhuber/tailwind-class-does-not-exist-issue/pull/1

Weird one!

**Reply by danielbachhuber** [reply]
You're the best 😊 Thanks @adamwathan !
